File: GuiaApe06/Ejercicio10.java
import java.util.LinkedList; // Librería para usar LinkedList
import java.util.Scanner;    // Librería para entrada de datos

// Clase Cliente
class Cliente {

    // Atributos privados
    private int turno;
    private String nombre;

    // Constructor de la clase
    public Cliente(int turno, String nombre) {

        this.turno = turno;
        this.nombre = nombre;
    }

    // Método para obtener el turno
    public int getTurno() {

        return turno;
    }

    // Método para obtener el nombre
    public String getNombre() {

        return nombre;
    }
}

// Clase para manejar la fila de atención
class FilaAtencion {

    // LinkedList para almacenar clientes
    private LinkedList<Cliente> fila;

    // Variable para controlar turnos
    private int siguienteTurno;

    // Constructor de la clase
    public FilaAtencion() {

        fila = new LinkedList<>();
        siguienteTurno = 1;
    }

    // Método para agregar clientes
    public void agregarCliente(String nombre) {

        // Crear nuevo cliente
        Cliente cliente = new Cliente(siguienteTurno, nombre);

        // Agregar cliente al final de la fila
        fila.addLast(cliente);

        // Mostrar turno asignado
        System.out.println("Cliente agregado con turno #" + siguienteTurno);

        // Incrementar siguiente turno
        siguienteTurno++;
    }

    // Método para atender clientes
    public void atenderCliente() {

        // Verificar si la fila está vacía
        if (fila.isEmpty()) {

            System.out.println("No hay clientes para atender.");
            return;
        }

        // Obtener primer cliente
        Cliente cliente = fila.getFirst();

        // Mostrar cliente atendido
        System.out.println("Atendiendo a "
                + cliente.getNombre()
                + " con turno #"
                + cliente.getTurno());

        // Eliminar cliente atendido
        fila.removeFirst();
    }

    // Método para mostrar la fila
    public void mostrarFila() {

        // Verificar si la fila está vacía
        if (fila.isEmpty()) {

            System.out.println("No hay clientes en espera.");
            return;
        }

        // Mostrar título
        System.out.println("\nFila de atencion:");

        // Recorrer y mostrar clientes
        for (Cliente cliente : fila) {

            System.out.println("Turno #"
                    + cliente.getTurno()
                    + " - "
                    + cliente.getNombre());
        }
    }
}

// Clase principal
public class Ejercicio10 {

    public static void main(String[] args) {

        // Crear objeto Scanner
        Scanner sc = new Scanner(System.in);

        // Crear objeto de la fila de atención
        FilaAtencion fila = new FilaAtencion();

        // Variable para el menú
        int opcion;

        // Ciclo principal
        do {

            // Mostrar menú
            System.out.println("\n===== FILA DE ATENCION =====");
            System.out.println("1. Agregar cliente");
            System.out.println("2. Atender cliente");
            System.out.println("3. Mostrar fila");
            System.out.println("4. Salir");

            // Pedir opción
            System.out.print("Seleccione una opcion: ");
            opcion = sc.nextInt();

            // Limpiar buffer
            sc.nextLine();

            // Evaluar opción seleccionada
            switch (opcion) {

                // Agregar cliente
                case 1:

                    System.out.print("Ingrese el nombre del cliente: ");
                    String nombre = sc.nextLine();

                    fila.agregarCliente(nombre);

                    break;

                // Atender cliente
                case 2:

                    fila.atenderCliente();

                    break;

                // Mostrar fila
                case 3:

                    fila.mostrarFila();

                    break;

                // Salir del programa
                case 4:

                    System.out.println("Saliendo...");
                    break;

                // Opción inválida
                default:

                    System.out.println("Opcion invalida.");
            }

        // Repetir hasta elegir salir
        } while (opcion != 4);

        // Cerrar Scanner
        sc.close();
    }
}  

File: GuiaApe06/Ejercicio10.cpp
#include <iostream>   // Librería para entrada y salida de datos
#include <list>       // Librería para usar listas
#include <string>     // Librería para usar cadenas de texto

using namespace std;

// Clase Cliente
class Cliente {

private:

    // Atributos privados
    int turno;
    string nombre;

public:

    // Constructor de la clase
    Cliente(int t = 0, string n = "") {

        turno = t;
        nombre = n;
    }

    // Método para obtener el turno
    int getTurno() const {

        return turno;
    }

    // Método para obtener el nombre
    string getNombre() const {

        return nombre;
    }
};

// Función para agregar clientes a la fila
void agregarCliente(list<Cliente>& fila, int& siguienteTurno) {

    // Variable para guardar nombre
    string nombre;

    // Pedir nombre del cliente
    cout << "Ingrese el nombre del cliente: ";
    cin >> nombre;

    // Agregar cliente a la lista
    fila.push_back(Cliente(siguienteTurno, nombre));

    // Mostrar turno asignado
    cout << "Cliente agregado con turno #" << siguienteTurno << endl;

    // Incrementar siguiente turno
    siguienteTurno++;
}

// Función para mostrar la fila
void mostrarFila(const list<Cliente>& fila) {

    // Verificar si la fila está vacía
    if (fila.empty()) {

        cout << "No hay clientes en espera.\n";
        return;
    }

    // Mostrar título
    cout << "\nFila de atencion:\n";

    // Recorrer y mostrar clientes
    for (const Cliente& c : fila) {

        cout << "Turno #" << c.getTurno()
             << " - " << c.getNombre() << endl;
    }
}

// Función para atender clientes
void atenderCliente(list<Cliente>& fila) {

    // Verificar si la fila está vacía
    if (fila.empty()) {

        cout << "No hay clientes para atender.\n";
        return;
    }

    // Mostrar cliente atendido
    cout << "Atendiendo a "
         << fila.front().getNombre()
         << " con turno #"
         << fila.front().getTurno() << endl;

    // Eliminar cliente atendido
    fila.pop_front();
}

int main() {

    // Crear lista de clientes
    list<Cliente> fila;

    // Variable para controlar turnos
    int siguienteTurno = 1;

    // Variable para opción del menú
    int opcion;

    // Ciclo principal del programa
    do {

        // Mostrar menú
        cout << "\n===== FILA DE ATENCION =====\n";
        cout << "1. Agregar cliente\n";
        cout << "2. Atender cliente\n";
        cout << "3. Mostrar fila\n";
        cout << "4. Salir\n";

        // Pedir opción
        cout << "Seleccione una opcion: ";
        cin >> opcion;

        // Evaluar opción seleccionada
        switch (opcion) {

            // Agregar cliente
            case 1:

                agregarCliente(fila, siguienteTurno);
                break;

            // Atender cliente
            case 2:

                atenderCliente(fila);
                break;

            // Mostrar fila
            case 3:

                mostrarFila(fila);
                break;

            // Salir del programa
            case 4:

                cout << "Saliendo...\n";
                break;

            // Opción inválida
            default:

                cout << "Opcion invalida.\n";
        }

    // Repetir hasta elegir salir
    } while (opcion != 4);

    return 0;
}
